### DISCUSSION TITLE
Automatically pass SOURCE_DATE_EPOCH to containers

### DISCUSSION THREAD

**Discussion by dalcinl** [pypa/cibuildwheel#1584] [Ideas]
Many build tools downstream cibuildwheel have support for reproducible builds via [`SOURCE_DATE_EPOCH`](https://reproducible-builds.org/specs/source-date-epoch/). 

Would it make sense passing this environment variable automatically to Linux containers (or any other OS containers cibuildwheel may support in the future)?

**Comment by joerick** [comment]
Yeah, I'd be interested in this. What would make sense for us, I wonder? Typically file mtimes in CI environments have the time of git checkout IIRC, but I suppose the right (reproducible) value would be the timestamp of the git commit that we're building.

That introduces a coupling to git, which we don't have currently. But we could easily add it to the cibuildwheel github action.

**Reply by dalcinl** [reply]
@joerick I'm not asking for cibuildwheel to determine (via git or whatever) and set the `SOURCE_DATE_EPOCH` environment variable. I'm just asking whether to pass it down to Linux containers. This is something that cibuildwheel supports if users *explitily* list `SOURCE_DATE_EPOCH` within the `environment-pass` option. I'm just asking to make this process automatic.

**Reply by joerick** [reply]
ah, right. Yeah, sounds good to me!

We could just modify this to add passthrough of SOURCE_DATE_EPOCH via the --env option to the container engine.

https://github.com/pypa/cibuildwheel/blob/18ff2c57c2558bd401d8c2639cc4271761f332e3/cibuildwheel/oci_container.py#L113-L128

**Reply by dalcinl** [reply]
@joerick Implementation looks indeed trivial. Where would you suggest this new feature should be documented?

**Reply by joerick** [reply]
A note in the environment-pass documentation would be sufficient, I think. https://cibuildwheel.readthedocs.io/en/stable/options/#environment-pass
